docker/build-tools/build-and-push.sh

```diff
@@ -17,13 +17,15 @@
 # support other container tools, e.g. podman
 CONTAINER_CLI=${CONTAINER_CLI:-docker}
 
+REPO=https://github.com/istio/tools.git
 HUB=${HUB:-gcr.io/istio-testing}
 DATE=$(date +%Y-%m-%dT%H-%M-%S)
 BRANCH=master
 VERSION="${BRANCH}-${DATE}"
+SHA=$(git ls-remote ${REPO} ${BRANCH} | cut -f 1)
```

Comment:
```suggestion
SHA=$(git rev-parse HEAD)
```

Reply:
I am changing this to `SHA=$(git rev-parse ${BRANCH})` to keep it consistent.

Reply:
Why? If we do that we need to manually maintain the branch, and it will give the same result as HEAD

Reply:
Martin had the concern here: https://github.com/istio/tools/pull/525. This is in consideration of local builds. I think it should be fine to assume that the script will only be run in CI system, so using `git rev-parse` is okay. wdyt?

Reply:
That PR was different since it grabs the branch. I think if you do `go get istio.io/tools@some-sha-only-local` it will work

Reply:
Oh actually it won't since its in docker :slightly_frowning_face: